From d3abe5c8d6a3711719c15efcecd2f77e7cc5142b Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Fri, 30 Aug 2019 14:14:00 +0200
Subject: [PATCH 14/23] UBUNTU: SAUCE: shiftfs: mark slab objects
 SLAB_RECLAIM_ACCOUNT

BugLink: https://bugs.launchpad.net/bugs/1842059

Shiftfs does not mark it's slab cache as reclaimable. While this is not
a big deal it is not nice to the kernel in general. The shiftfs cache is
not so important that it can't be reclaimed.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
Acked-by: Seth Forshee <seth.forshee@canonical.com>
Acked-by: Stefan Bader <stefan.bader@canonical.com>
Signed-off-by: Kleber Sacilotto de Souza <kleber.souza@canonical.com>
---
 fs/shiftfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/shiftfs.c b/fs/shiftfs.c
index 9006201c243d..5aaef6e703ce 100644
--- a/fs/shiftfs.c
+++ b/fs/shiftfs.c
@@ -2103,7 +2103,7 @@ static int __init shiftfs_init(void)
 {
 	shiftfs_file_info_cache = kmem_cache_create(
 		"shiftfs_file_info_cache", sizeof(struct shiftfs_file_info), 0,
-		SLAB_HWCACHE_ALIGN | SLAB_ACCOUNT | SLAB_MEM_SPREAD, NULL);
+		SLAB_RECLAIM_ACCOUNT | SLAB_HWCACHE_ALIGN | SLAB_ACCOUNT | SLAB_MEM_SPREAD, NULL);
 	if (!shiftfs_file_info_cache)
 		return -ENOMEM;
 
-- 
2.20.1

